### PR TITLE
LOK-2477: New Syslog Config task set and publish to Minion.

### PR DESCRIPTION
**Pull request by MusaidAli**
## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2477

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        publishSyslogConfig(tenantId, locationId, syslogConfig);
+    }
+
+    private SyslogConfig mapBeanToProto(SyslogConfigBean config) {
+        return SyslogConfig.newBuilder()
+                .setSyslogAddress(config.getSyslogAddress())
+                .setSyslogPort(config.getSyslogPort())
+                .setNewSuspectOnSyslog(config.getNewSuspectOnSyslog())
+                .setIncludeRawMessage(config.isIncludeRawMessage())
+                //            .setUseAddressFromVarbind(config.shouldUseAddressFromVarbind())
```
Comment (fooker) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java:90]: Please remove

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    private Integer queueSize;
+
+    @JsonProperty("numThreads")
+    private Integer numThreads;
+
+    @JsonProperty("batchIntervalMs")
+    private Integer batchIntervalMs;
+
+    @JsonIgnore
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
```
Comment (fooker) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java:73]: What are these used for?

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto [hunk 3]
```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+package opennms.sink.syslog;
+option java_multiple_files = true;
+option java_package = "org.opennms.sink.syslog.contract";
+
+message SyslogConfig {
+   string syslog_address = 1;
```
Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto:10]: Format

**Review by fooker — CHANGES_REQUESTED** [review]

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java (hunk 1 above)
Comment (MusaidAli) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java]: removed.

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto (hunk 3 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog-config.proto]: formatted .

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java (hunk 2 above)
Comment (MusaidAli) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/syslogconfig/SyslogConfigBean.java:71]: removed ,no idea it is auto generated i think.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setNameFormat("syslog-config-update-scheduler-%d")
+            .build();
+    private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void scheduleConfigUpdate() {
+        // This is work around for Minion not to have timeout for CloudToMinion Stream.
+        executorService.scheduleAtFixedRate(this::sendSyslogConfigToMinionAfterStartup, 900, 900, TimeUnit.SECONDS);
```
Comment (smunir-onms) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java:67]: suggestion, 900, 900 should be configured values

File: inventory/main/src/test/java/org/opennms/horizon/inventory/service/SyslogConfigServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+                        .setTenantId(TEST_TENANT_ID)
+                        .setId(TEST_LOCATION_ID2)
+                        .setLocation(TEST_LOCATION_ID2_NAME)
+                        .build());
+        Mockito.when(mockMonitoringLocationService.findAll()).thenReturn(testLocationList);
+
+        /*      var testException = new RuntimeException("x-test-exception-x");
+        Mockito.doThrow(testException)
+            .when(mockTaskSetPublisher)
+            .publishNewTasks(Mockito.anyString(), Mockito.eq(TEST_LOCATION_ID1), Mockito.any(List.class));*/
```
Comment (smunir-onms) [inline inventory/main/src/test/java/org/opennms/horizon/inventory/service/SyslogConfigServiceTest.java:120]: remove commented code

Comment (MusaidAli) [reply inventory/main/src/test/java/org/opennms/horizon/inventory/service/SyslogConfigServiceTest.java]: resolved.removed commented code.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java (hunk 4 above)
Comment (MusaidAli) [reply inventory/main/src/main/java/org/opennms/horizon/inventory/service/SyslogConfigService.java]: resolved.values from config.

**Review by fooker — APPROVED** [review]

**Comment by fooker** [comment]
Looks good to me. Tests still failing

**Pull request merged**